File: app/src/main/java/net/veldor/flibustaloader/MyWebView.java
package net.veldor.flibustaloader;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.webkit.WebChromeClient;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.widget.ProgressBar;

import net.veldor.flibustaloader.ui.WebViewActivity;

public class MyWebView extends WebView {

    private boolean init = false;

    public MyWebView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }


    @SuppressLint("SetJavaScriptEnabled")
    public void setup() {
        if (!this.isInEditMode()) {
            this.setWebViewClient(new MyWebViewClient());
            WebSettings webSettings = this.getSettings();
            webSettings.setJavaScriptEnabled(true);
            webSettings.setAllowFileAccess(true);
            webSettings.setBuiltInZoomControls(true);
            webSettings.setDisplayZoomControls(false);
        }
    }

    @Override
    public void loadUrl(String url) {
        super.loadUrl(url);
        initProgressBar();
    }

    private void initProgressBar() {
        if (init) {
            return;
        }
        init = true;
        final ProgressBar progressBar = ((WebViewActivity) getContext()).findViewById(R.id.pageLoadedProgressBar);
        // попробую скрыть бар для начала
        progressBar.setVisibility(View.GONE);
        this.setWebChromeClient(new WebChromeClient() {
            public void onProgressChanged(WebView view, int progress) {
                if (progress > 90) {
                    if (progressBar.getVisibility() == View.VISIBLE) {
                        progressBar.setVisibility(View.GONE);
                    }
                } else {
                    if (progressBar.getVisibility() == View.GONE) {
                        progressBar.setVisibility(View.VISIBLE);
                    }
                    progressBar.setProgress(progress);
                }
            }
        });
    }
}

File: app/src/main/java/net/veldor/flibustaloader/MyWebView.kt
package net.veldor.flibustaloader

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.webkit.WebChromeClient
import android.webkit.WebView
import net.veldor.flibustaloader.ui.BrowserActivity
import net.veldor.flibustaloader.ui.fragments.WebViewFragment
import net.veldor.flibustaloader.utils.URLHelper

class MyWebView(context: Context?, attrs: AttributeSet?) : WebView(context, attrs) {
    private var init = false

    @SuppressLint("SetJavaScriptEnabled")
    fun setup() {
        if (!this.isInEditMode) {
            this.webViewClient = MyWebViewClient()
            val webSettings = this.settings
            webSettings.javaScriptEnabled = true
            webSettings.allowFileAccess = true
            webSettings.builtInZoomControls = true
            webSettings.displayZoomControls = false
        }
    }


    override fun loadUrl(url: String) {
        super.loadUrl(URLHelper.getBaseUrl() + url)
        initProgressBar()
    }

    private fun initProgressBar() {
        if (init) {
            return
        }
        init = true
        val fragment = (context as BrowserActivity).getCurrentFragment()
        if (fragment is WebViewFragment) {
            fragment.binding.pageLoadedProgressBar.visibility = GONE
            this.webChromeClient = object : WebChromeClient() {
                override fun onProgressChanged(view: WebView, progress: Int) {
                    if (progress > 90) {
                        if (fragment.binding.pageLoadedProgressBar.visibility == VISIBLE) {
                            fragment.binding.pageLoadedProgressBar.visibility = GONE
                        }
                    } else {
                        if (fragment.binding.pageLoadedProgressBar.visibility == GONE) {
                            fragment.binding.pageLoadedProgressBar.visibility = VISIBLE
                        }
                        fragment.binding.pageLoadedProgressBar.progress = progress
                    }
                }
            }
        }
    }
}